src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs

```diff
@@ -923,22 +925,38 @@ private PointsToAbstractValue VisitInvocationCommon(IOperation operation, IOpera
                 return value;
             }
 
-            private static bool IsSpecialEmptyOrFactoryMethod(IMethodSymbol method)
-                => IsSpecialFactoryMethod(method) || IsSpecialEmptyMember(method);
+            private static bool IsSpecialMethodReturningNonNullValue(IMethodSymbol method, WellKnownTypeProvider wellKnownTypeProvider)
+                => IsSpecialFactoryMethodReturningNonNullValue(method, wellKnownTypeProvider) || IsSpecialEmptyMember(method);
 
             /// <summary>
             /// Returns true if this special static factory method whose name starts with "Create", such that
+            /// method's return type is not nullable, i.e. 'type?', and
             /// method's containing type is static OR a special type OR derives from or is same as the type of the field/property/method return.
             /// For example: class SomeType { static SomeType CreateXXX(...); }
             /// </summary>
-            private static bool IsSpecialFactoryMethod(IMethodSymbol method)
+            private static bool IsSpecialFactoryMethodReturningNonNullValue(IMethodSymbol method, WellKnownTypeProvider wellKnownTypeProvider)
             {
-                return method.IsStatic &&
+                var isFactoryMethod = method.IsStatic &&
```

Comment:
Could we split this condition? It's pretty complex to read with all the mix of AND and OR conditions.

Reply:
Sure, done.